stats.go

```diff
@@ -0,0 +1,1306 @@
+	// this Stats object. Two Stats objects will have the same ID if they were produced
+	// by inspecting the same underlying object.
+	ID string `json:"id"`
+}
+
+// CodecType specifies whether a CodecStats objects represents a media format
+// that is being encoded or decoded
+type CodecType string
+
+const (
+	// CodecTypeEncode means the attached CodecStats represents a media format that
+	// is being encoded, or that the implementation is prepared to encode.
+	CodecTypeEncode CodecType = "encode"
+	// CodecTypeDecode means the attached CodecStats represents a media format
+	// that the implementation is prepared to decode.
+	CodecTypeDecode CodecType = "decode"
+)
+
+// CodecStats contains statistics for a codec that is currently being used by RTP streams
+// being sent or received by this PeerConnection object.
+type CodecStats struct {
+	StatsBase
+	// PayloadType as used in RTP encoding or decoding
+	PayloadType uint32 `json:"payloadType"`
+	// CodecType of this CodecStats
+	CodecType CodecType `json:"codecType"`
+	// TransportID is the unique identifier of the transport on which this codec is
+	// being used, which can be used to look up the corresponding TransportStats object.
+	TransportID string `json:"transportId"`
+	// MimeType is the codec MIME media type/subtype. e.g., video/vp8 or equivalent.
+	MimeType string `json:"mimeType"`
+	// ClockRate represents the media sampling rate.
+	ClockRate uint32 `json:"clockRate"`
+	// Channels is 2 for stereo, missing for most other cases.
+	Channels uint32 `json:"channels"`
+	// SDPFmtpLine is the a=fmtp line in the SDP corresponding to the codec,
+	// i.e., after the colon following the PT.
+	SDPFmtpLine string `json:"sdpFmtpLine"`
+	// Implementation identifies the implementation used. This is useful for diagnosing
+	// interoperability issues.
+	Implementation string `json:"implementation"`
+}
+
+// Timestamp returns the timestamp associated with this object.
+//
+// For statistics that came from a remote source (e.g., from received RTCP packets),
+// timestamp represents the time at which the information arrived at the local endpoint.
+func (s CodecStats) Timestamp() StatsTimestamp {
+	return s.StatsBase.Timestamp
+}
+
+// Type returns the object's StatsType
+func (s CodecStats) Type() StatsType {
+	return s.StatsBase.Type
+}
+
+// ID returns a unique id that is associated with the component inspected to produce
+// this Stats object. Two Stats objects will have the same ID if they were produced
+// by inspecting the same underlying object.
+func (s CodecStats) ID() string {
+	return s.StatsBase.ID
+}
+
+// ReceivedRTPStreamStats contains stats measured at the receiving end of an RTP stream,
+// known either because they're measured locally or transmitted via an RTCP Receiver
+// Report (RR) or Extended Report (XR).
+type ReceivedRTPStreamStats struct {
+	// PacketsReceived is the total number of RTP packets received for this SSRC.
+	PacketsReceived uint32 `json:"packetsReceived"`
```

Comment:
NIT -- What do you think of putting a newline after every member? Visually I break chunks apart by newline, so easier to consume when reading

Reply:
will do